File: Ruben - Modulo 2/prAltura/src/alturas/Paises.java
package alturas;

import java.io.File;
import java.io.FileNotFoundException;
import java.lang.reflect.Array;
import java.util.*;

public class Paises {
    private List<Pais> paises;

    public Paises() {
        paises = new ArrayList<>();
    }

    public List<Pais> getPaises() {
        return paises;
    }

    public void leePaises(String file) throws FileNotFoundException {
        try (Scanner sc = new Scanner(new File(file))) {
            leePaises(sc);
        }
    }

    private void leePaises(Scanner sc) {
        int nPaises = 0;

        while (sc.hasNextLine()) {
            try (Scanner scPais = new Scanner(sc.nextLine()).useLocale(Locale.ENGLISH)) {
                scPais.useDelimiter(",");
                String nombre = scPais.next();
                String continente = scPais.next();
                double altura = scPais.nextDouble();
                paises.add(new Pais(nombre, continente, altura));
            }
        }
    }

    public List<Pais> selecciona(Seleccion sel){
        List<Pais> res = new ArrayList<>();

        for (Pais pais : paises) {
            if (sel.test(pais)){
                res.add(pais);
            }
        }

        return res;
    }
}

File: Ruben - Modulo 2/prAlturaKT/src/alturas/Paises.kt
package alturas

import java.io.File
import java.io.FileNotFoundException
import java.util.ArrayList
import java.util.Locale
import java.util.Scanner

class Paises {
    val paises: MutableList<Pais>

    init {
        paises = ArrayList()
    }

    @Throws(FileNotFoundException::class)
    fun leePaises(file: String) {
        Scanner(File(file)).use { sc -> leePaises(sc) }
    }

    private fun leePaises(sc: Scanner) {
        val nPaises = 0

        while (sc.hasNextLine()) {
            Scanner(sc.nextLine()).useLocale(Locale.ENGLISH).use { scPais ->
                scPais.useDelimiter(",")
                val nombre = scPais.next()
                val continente = scPais.next()
                val altura = scPais.nextDouble()
                paises.add(Pais(nombre, continente, altura))
            }
        }
    }

    fun selecciona(sel: Seleccion): List<Pais> {
        val res = ArrayList<Pais>()

        for (pais in paises) {
            if (sel.test(pais)) {
                res.add(pais)
            }
        }

        return res
    }
}
